.github/workflows/publish-main.yaml

```diff
@@ -0,0 +1,75 @@
+name: Publish to main 
+
+on:
+  push:
+    branches:
+      - 'Branch Name'
+  
+env:
+  GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }} 
+  release-version: "0.0.0"
+  community_operators_dir_path: "../community-operators/operators/gingersnap/"
+  community_operators_prod_dir_path: "../community-operators-prod/operators/gingersnap/"
+
+  
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Configure Git
+        run: |
+          git config --global user.name "pminz"
+          git config --global user.email "pminz@redhat.com"
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with: 
+           path: operator
+
+
+      - name: Checkout community-operators repo
+        uses: actions/checkout@v2
+        with:
+          repository: k8s-operatorhub/community-operators
+          path: community-operators
+          ref: main
+          token: ${{ secrets.PAT_TOKEN }}
+
+      - name: Check if new version directory exists in community-operators repo
```

Comment:
Thinking about this more, I don't think this is sufficient as there's a delay between a community-operators PR being merged and OLM bundle image being available on OperatorHub.

Instead, we can use docker to check whether the required image has been published yet:

```
docker manifest inspect quay.io/operatorhubio/gingersnap:v${{ env.release-version}} > /dev/null
```

^ will return an exit code of 1 if the image doesn't exist yet, causing the step to fail.